scripts/install_environment.sh

```diff
@@ -1,45 +1,126 @@
+
+  if ! version_is_greater "$current_version" "$minimum_version"; then
+    echo "Adding --listen flag to libvirt"
+    sudo sed -i -e 's/#LIBVIRTD_ARGS="--listen"/LIBVIRTD_ARGS="--listen"/g' /etc/sysconfig/libvirtd
+    sudo systemctl restart libvirtd
   else
-    echo "libvirt is already installed"
+    echo "libvirtd version is greater then 5.5.x, starting libvirtd-tcp.socket"
+    sudo systemctl stop libvirtd
+    sudo systemctl restart libvirtd.socket
+    sudo systemctl enable --now libvirtd-tcp.socket
+    sudo systemctl start libvirtd-tcp.socket
+    sudo systemctl start libvirtd
   fi
-  
-  sed -i -e 's/#LIBVIRTD_ARGS="--listen"/LIBVIRTD_ARGS="--listen"/g' /etc/sysconfig/libvirtd
-  sed -i -e 's/#listen_tls/listen_tls/g' /etc/libvirt/libvirtd.conf
-  sed -i -e 's/#listen_tcp/listen_tcp/g' /etc/libvirt/libvirtd.conf
-  sed -i -e 's/#auth_tcp = "sasl"/auth_tcp = "none"/g' /etc/libvirt/libvirtd.conf
-  sed -i -e 's/#tcp_port/tcp_port/g' /etc/libvirt/libvirtd.conf
-  sed -i -e 's/#security_driver = "selinux"/security_driver = "none"/g' /etc/libvirt/qemu.conf
+  if sudo virsh net-list --all | grep default | grep  inactive; then
+      echo "default network is inactive, fixing it"
+      sudo ip link del virbr0-nic
+      sudo ip link del virbr0
+      sudo virsh net-start default
+  fi
+  current_user=$(whoami)
+  echo "Adding user $current_user ti libvirt and qemu groups"
+  sudo gpasswd -a $current_user libvirt
+  sudo gpasswd -a $current_user qemu
+
 }
 
 function install_runtime_container() {
+  echo "Installing container runitme package"
   if ! [ -x "$(command -v docker)" ] && ! [ -x "$(command -v podman)" ]; then
-    dnf install podman-1.6.4 -y
+    sudo dnf install podman -y
   elif [ -x "$(command -v podman)" ]; then
-    dnf install podman-1.6.4 -y
+    current_version="$(podman -v | awk '{print $3}')"
+    minimum_version="1.6.4"
+    if ! version_is_greater "$current_version" "$minimum_version"; then
+      sudo dnf install podman-$minimum_version  -y
+    fi
   else
     echo "docker or podman is already installed"
   fi
 }
 
 function install_packages(){
-  dnf install -y make python3 git jq bash-completion xinetd
-  systemctl enable --now xinetd
+  echo "Installing dnf packages"
+  sudo dnf install -y make python3 python3-pip git jq bash-completion xinetd
+  sudo systemctl enable --now xinetd
 }
 
 function install_skipper() {
-   pip3 install strato-skipper==1.20.0
+   echo "Installing skipper and adding ~/.local/bin to PATH"
+   pip3 install strato-skipper==1.22.0 --user
+   sudo cp ~/.local/bin/skipper /usr/local/bin
+   # TODO maybe better to add ,local to PATH
+   # grep -qxF "export PATH=~/.local/bin:$PATH" ~/.bashrc || echo "export PATH=~/.local/bin:$PATH" >> ~/.bashrc
+   # export PATH="$PATH:~/.local/bin"
 }
 
+function config_firewalld() {
+  echo "Config firewall"
+  sudo systemctl start firewalld
+  if  [ "${NO_EXTERNAL_PORT}" = "n" ];then
```

Comment:
The name of the variables confuses. I don't like actually "no" in my variables.
Change to "EXTERNAL_PORT" with a default of "y"